core/editor/src/tool/tools/fill.rs

```diff
@@ -0,0 +1,34 @@
+use crate::message_prelude::*;
+use crate::tool::ToolActionHandlerData;
+use document_core::Operation;
+
+#[derive(Default)]
+pub struct Fill;
+
+#[impl_message(Message, ToolMessage, Fill)]
+#[derive(PartialEq, Clone, Debug)]
+pub enum FillMessage {
+	MouseDown,
+}
+
+impl<'a> MessageHandler<ToolMessage, ToolActionHandlerData<'a>> for Fill {
+	fn process_action(&mut self, action: ToolMessage, data: ToolActionHandlerData<'a>, responses: &mut VecDeque<Message>) {
+		let mouse_pos = data.2.mouse.position;
+		let quad = [
```

Comment:
You could take a look at `src/tool/tools/select.rs` for how to genericise selection tolerances.
It would be great if you could also put the selection tolerance const into the `src/consts.rs` module
oh wait, nevermind that does not exist yet but you could add the constant with a todo comment